### PR TITLE
Move `/info` outside authorization

### PR DESCRIPTION
**Pull request by Gusted**
- To use the web's API to get information about a issue/pull on a repository, doesn't require authorization(nor that the repository isn't archived).
- Regressed by: #19318

### CODE REVIEW

**Comment by lunny** [comment]
> * Never use /api/v1 from Gitea UI Pages #19318

1 Why do you think it doesn't need authorization? What about a private repository's issues?
2 routes in Web should only be visited from UI. Gitea should not guarantee the routes path compatibility and stability.

**Comment by Gusted** [comment]
> 1 Why do you think it doesn't need authorization? What about a private repository's issues?

Because not every user on a Gitea instance would be logged in and they shouldn't be shown a loading spinner when hovering over a issue reference. Private repository's issues are not being checked in `reqSignin`, but in `context.RepoAssignment` IIRC, that's the main difference.





> 2 routes in Web should only be visited from UI. Gitea should not guarantee the routes path compatibility and stability.

It's still on the same path as before, now it just doesn't have the `reqSignin` middleware.

**Review by lunny — APPROVED** [review]

**Comment by zeripath** [comment]
Hmm... I think it should require permissions check. If you can't read the issues or pull requests you should not be able to access this endpoint. How does your change affect leakage for private repos where you have been given say code access but not read issues access?

**Comment by zeripath** [comment]
This also needs:

```patch
diff --git a/routers/web/repo/issue.go b/routers/web/repo/issue.go
index 079ccbf6c..875cf421e 100644
--- a/routers/web/repo/issue.go
+++ b/routers/web/repo/issue.go
@@ -1799,6 +1799,25 @@ func GetIssueInfo(ctx *context.Context) {
 		}
 		return
 	}
+
+	if issue.IsPull {
+		// Need to check if Pulls are enabled and we can read Pulls
+		if !ctx.Repo.Repository.CanEnablePulls() || !ctx.Repo.CanRead(unit.TypePullRequests) {
+			ctx.Error(http.StatusNotFound)
+			return
+		}
+	} else {
+		if !ctx.Repo.CanRead(unit.TypeIssues) &&
+			!ctx.Repo.CanRead(unit.TypeExternalTracker) {
+			ctx.Error(http.StatusNotFound)
+			return
+		}
+		_, err := ctx.Repo.Repository.GetUnit(unit.TypeExternalTracker)
+		if err == nil {
+			ctx.Error(http.StatusNotFound)
+		}
+	}
+
 	ctx.JSON(http.StatusOK, convert.ToAPIIssue(issue))
 }
 
```

**Comment by zeripath** [comment]
Realistic we might actually need a `context.IssueAssignment()` function that can do this sort of check as I bet that there are lots of other places where this check is not happening properly.

File: routers/web/repo/issue.go [hunk 1]
```diff
@@ -1799,6 +1799,27 @@ func GetIssueInfo(ctx *context.Context) {
...
+		if !ctx.Repo.CanRead(unit.TypeIssues) &&
+			!ctx.Repo.CanRead(unit.TypeExternalTracker) {
+			ctx.Error(http.StatusNotFound)
+			return
+		}
+		_, err := ctx.Repo.Repository.GetUnit(unit.TypeExternalTracker)
+		if err == nil {
+			ctx.Error(http.StatusNotFound)
+			return
+		}
```
Comment (zeripath) [inline routers/web/repo/issue.go]: Actually we can be even simpler here.

```suggestion
		if !ctx.Repo.CanRead(unit.TypeIssues)  {
			ctx.Error(http.StatusNotFound)
			return
		}
```

**Review by zeripath — APPROVED** [review]

**Pull request merged**